File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    def moveCrain(move, board):
        for i in range(len(board)):
            if board[i][move-1] == 0:
                continue
            else:
                return [i, move - 1, board[i][move-1]]
        return [-1, -1, 0]
    basket = []
    for move in moves:
        i, j, doll = moveCrain(move, board)
        if i == -1:
            continue
        else:
            board[i][j] = 0
            if len(basket) == 0:
                basket.append(doll)
            else:
                if basket[-1] == doll:
                    basket.pop()
                    answer += 2
                else:
                    basket.append(doll)
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.swift
import Foundation

func solution(_ board:[[Int]], _ moves:[Int]) -> Int {
    func moveCrain (_ move: Int, _ board: [[Int]]) -> [Int]? {
        for i in 0..<board.count {
            if board[i][move - 1] == 0 {
                continue
            }
            else {
                return [i, move - 1, board[i][move - 1]]
            }
        }
        return nil
    }
    
    var answer = 0
    var board = board
    var basket = [Int]()
    for move in moves {
        if let result = moveCrain(move, board) {
            let i = result[0]
            let j = result[1]
            let doll = result[2]
            board[i][j] = 0
            if let last = basket.popLast() {
                if doll != last {
                    basket.append(last)
                    basket.append(doll)
                }
                else {
                    answer += 2
                }
            }
            else {
                basket.append(doll)
            }
        }
        else {
            continue
        }
    }
    
    return answer
}